File: listas/src/main/java/com/victorlh/android/framework/listas/adapter/filtros/ListaFiltrable.java
package com.victorlh.android.framework.listas.adapter.filtros;

import com.victorlh.android.framework.listas.ItemLista;

public interface ListaFiltrable<T extends ItemLista> {

	void filtrar();

	void addFiltro(FiltroLista<T> filtro);

	void removeFiltro(FiltroLista<T> filtro);

	void removeAllFiltros();

	void invalidate();
}

File: listas/src/main/java/com/victorlh/android/framework/listas/adapter/filtros/ListaFiltrable.kt
package com.victorlh.android.framework.listas.adapter.filtros

interface ListaFiltrable<T> {
    fun filtrar()
    fun addFiltro(filtro: FiltroLista<T>)
    fun removeFiltro(filtro: FiltroLista<T>)
    fun removeAllFiltros()
    fun invalidate()
}